admin/taxonomy/class-taxonomy-metabox.php

```diff
@@ -54,144 +56,118 @@ class WPSEO_Taxonomy_Metabox {
 	public function __construct( $taxonomy, $term ) {
 		$this->term                 = $term;
 		$this->taxonomy             = $taxonomy;
-		$this->taxonomy_tab_content = new WPSEO_Taxonomy_Fields_Presenter( $this->term );
+		$this->social_is_enabled = WPSEO_Options::get( 'opengraph', false ) || WPSEO_Options::get( 'twitter', false );
+
+		$this->seo_analysis = new WPSEO_Metabox_Analysis_SEO();
+		$this->readability_analysis = new WPSEO_Metabox_Analysis_Readability();
 	}
 
 	/**
 	 * Shows the Yoast SEO metabox for the term.
 	 */
 	public function display() {
-
-		$content_sections = $this->get_content_sections();
-
-		$product_title = 'Yoast SEO';
-		if ( file_exists( WPSEO_PATH . 'premium/' ) ) {
-			$product_title .= ' Premium';
-		}
-
 		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Reason: $product_title is hardcoded.
-		printf( '<div id="wpseo_meta" class="postbox yoast wpseo-taxonomy-metabox-postbox"><h2><span>%1$s</span></h2>', $product_title );
+		printf( '<div id="wpseo_meta" class="postbox yoast wpseo-taxonomy-metabox-postbox"><h2><span>%1$s</span></h2>', $this->get_product_title() );
 
 		echo '<div class="inside">';
 		echo '<div id="taxonomy_overall"></div>';
 
+		$this->render_hidden_fields();
+		$this->render_tabs();
 
-		echo '<div class="wpseo-metabox-content">';
-		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- Reason: $product_title is hardcoded.
-		printf( '<div class="wpseo-metabox-menu"><ul role="tablist" class="yoast-aria-tabs" aria-label="%s">', $product_title );
-
-		foreach ( $content_sections as $content_section ) {
-			$content_section->display_link();
-		}
+		echo '</div>';
+		echo '</div>';
+	}
 
-		echo '</ul></div>';
+	/**
+	 * Renders the metabox hidden fields.
+	 *
+	 * @return void
+	 */
+	protected function render_hidden_fields() {
```

Comment:
A general comment about this function, we are doing some checks double (like whether social is enabled, because `WPSEO_Taxonomy_Fields` already has social checks), some checks we only do here (for settings) and some we only do in  `WPSEO_Taxonomy_Fields` (breadcrumbs). Maybe it's a good idea to do these checks in one of two places.

Reply:
Valid concern, but let's save that for later. This PR is one in a long range of steps towards DRYing those integrations where possible.